_posts/2021-10-25-waku-v1-vs-waku-v2.md

```diff
@@ -0,0 +1,202 @@
+This necessarily leads to unnecessary duplication (termed _amplification factor_),
+wasting bandwidth and resources.
+What's more, we expect this effect to worsen the larger the network becomes,
+as each _connection_ will receive a copy of each message,
+rather than a single copy per peer.
+
+Message routing in Waku v2 follows the `libp2p` _GossipSub_ protocol,
+which lowers amplification factors by only sending full message contents to a subset of connected peers.
+As a Waku v2 network grows, each peer will limit its number of full-message ("mesh") peerings -
+`libp2p` suggests a maximum of `12` such connections per peer.
+This allows much better scalability than a flood-routed network.
+From time to time, a Waku v2 peer will send metadata about the messages it has seen to other peers ("gossip" peers).
+
+See [this explainer](https://hackmd.io/@vac/main/%2FYYlZYBCURFyO_ZG1EiteWg#11WAKU2-RELAY-gossipsub) for a more detailed discussion.
+
+## Methodology
+
+The results below contain only some scenarios that provide an interesting contrast between Waku v1 and Waku v2.
+For example, [star network topologies](https://www.techopedia.com/definition/13335/star-topology#:~:text=Star%20topology%20is%20a%20network,known%20as%20a%20star%20network.) do not show a substantial difference between Waku v1 and Waku v2.
+This is because each peer relies on a single connection to the central node for every message,
+which barely requires any routing:
+each connection receives a copy of every message for both Waku v1 and Waku v2.
+Hybrid topologies similarly show only a difference between Waku v1 and Waku v2 for network segments with [mesh-like connections](https://en.wikipedia.org/wiki/Mesh_networking),
+where routing decisions need to be made.
+
+For this reason, the following approach applies to all scenarios:
+1. Simulations are run **locally**.
+This limits the size of possible scenarios due to local resource constraints,
+but is a way to quickly get an approximate comparison.
+2. Nodes are treated as a **blackbox** for which we only measure bandwidth,
+using an external bandwidth monitoring tool.
+In other words, we do not consider differences in the size of the envelope (for v1) or the message (for v2).
+3. Messages are published at a rate of **50 new messages per second** to each network,
+except where explicitly stated otherwise.
+4. Each message propagated in the network carries **8 bytes** of random payload, which is **encrypted**.
+The same symmetric key cryptographic algorithm (with the same keys) are used in both Waku v1 and v2.
+5. Traffic in each network is **generated from 10 nodes** (randomly-selected) and published in a round-robin fashion to **10 topics** (content topics for Waku v2).
+In practice, we found no significant difference in _average_ bandwidth usage when tweaking these two parameters (the number of traffic generating nodes and the number of topics).
```

Comment:
Ah, I understand your question a bit better now. Yes, if we were to normalise the bandwidth to _per message_ we would not see much variation if we also tweaked the number of nodes generating traffic or the number of content topics. However, the actual scenario results are certainly _per node_, which is heavily dependent on the message rate - the higher the message rate, the higher the average bandwidth per node.

Reply:
As a user of waku, I would be also interested to see the bandwidth overhead per node and per message i.e, I would ask how much bandwidth each node has to spend for the transmission of one message to the entire network. In another set of simulations, the impact of message rate on bandwidth can be examined for scenarios with identical setup but varying messaging rates.

In the current PR, since the objective is to provide a comparative analysis of waku v1 with v2, I think the average per node is fine. But for the follow-up simulations that you mentioned (to characterize waku v2), I'd suggest having per-message results as well.